zarr/creation.py

```diff
@@ -16,7 +16,7 @@
 
 
 def create(shape, chunks=True, dtype=None, compressor='default',
-           fill_value=0, order='C', store=None, synchronizer=None,
+           fill_value=None, order='C', store=None, synchronizer=None,
```

Comment:
Are there any other effects from changing this from `0` to `None`?

Should we be updating `fill_value`s elsewhere to match?

Reply:
The only other use of `fill_value=0` I can see is `creation.open_array`. I assume this should probably be `None` too, but I'm not too familiar with all the effects here.

Reply:
`open_array` just [calls `init_array` if needed]( https://github.com/zarr-developers/zarr-python/blob/f461eb78fbb88187582cd9123d6ec7622d9abd26/zarr/creation.py#L515-L539 ), which already has [`fill_value=None`]( https://github.com/zarr-developers/zarr-python/blob/22ded1d656cb2034180cf8a3f23f82b51f79d8b4/zarr/storage.py#L231 ). So making it `None` in `open_array` seems more consistent with the underlying function.